clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2537,97 +2610,120 @@ public DescribeConsumerGroupsResult describeConsumerGroups(final Collection<Stri
+                        .setGroups(Collections.singletonList(context.getGroupId()))
+                        .setIncludeAuthorizedOperations(context.getOptions().includeAuthorizedOperations()));
+            }
 
-                            if (handleGroupRequestError(groupError, future))
-                                return;
+            @Override
+            void handleResponse(AbstractResponse abstractResponse) {
+                final DescribeGroupsResponse response = (DescribeGroupsResponse) abstractResponse;
 
-                            final String protocolType = describedGroup.protocolType();
-                            if (protocolType.equals(ConsumerProtocol.PROTOCOL_TYPE) || protocolType.isEmpty()) {
-                                final List<DescribedGroupMember> members = describedGroup.members();
-                                final List<MemberDescription> memberDescriptions = new ArrayList<>(members.size());
-                                final Set<AclOperation> authorizedOperations = validAclOperations(describedGroup.authorizedOperations());
-                                for (DescribedGroupMember groupMember : members) {
-                                    Set<TopicPartition> partitions = Collections.emptySet();
-                                    if (groupMember.memberAssignment().length > 0) {
-                                        final PartitionAssignor.Assignment assignment = ConsumerProtocol.
-                                            deserializeAssignment(ByteBuffer.wrap(groupMember.memberAssignment()));
-                                        partitions = new HashSet<>(assignment.partitions());
-                                    }
-                                    final MemberDescription memberDescription =
-                                        new MemberDescription(groupMember.memberId(),
-                                            groupMember.clientId(),
-                                            groupMember.clientHost(),
-                                            new MemberAssignment(partitions));
-                                    memberDescriptions.add(memberDescription);
-                                }
-                                final ConsumerGroupDescription consumerGroupDescription =
-                                    new ConsumerGroupDescription(groupId, protocolType.isEmpty(),
-                                        memberDescriptions,
-                                        describedGroup.protocolData(),
-                                        ConsumerGroupState.parse(describedGroup.groupState()),
-                                        fcResponse.node(),
-                                        authorizedOperations);
-                                future.complete(consumerGroupDescription);
-                            }
-                        }
+                final DescribedGroup describedGroup = response.data()
```

Comment:
Since we're only using a batch size of one, we should be able to just verify that we got a list of size 1 and it contained what we wanted, right?  Also, we should also try to be robust against malformed responses that don't include the group we asked for (this will currently throw an NPE if we get one of those).

Reply:
Done. After adding the check one unit test failed, I have fixed it (need to delete some code) and added another unit test that incorporates the deleted code and tests it in correct way. Please take a look at that.